### ISSUE TITLE
openblas@0.3.21 w/ Cray PrgEnv-gnu: gfortran: noomp: linker input file not found: No such file or directory...

### ISSUE THREAD

**Issue by eugeneswalker** [LLNL/spack#33442]
### Steps to reproduce the issue

Having some trouble building `openblas@0.3.21 %gcc@11.2.0` using PrgEnv-gnu on OLCF Crusher.

Concretization:
```
 -   3p4uqpa  openblas@0.3.21%gcc@11.2.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared symbol_suffix=none threads=none arch=cray-sles15-zen3
[+]  74xxjm4      ^perl@5.34.1%gcc@11.2.0+cpanm+shared+threads arch=cray-sles15-zen3
[+]  o7ojbiz          ^berkeley-db@18.1.40%gcc@11.2.0+cxx~docs+stl patches=26090f4,b231fcc arch=cray-sles15-zen3
[+]  rll2prr          ^bzip2@1.0.8%gcc@11.2.0~debug~pic+shared arch=cray-sles15-zen3
[+]  3s2io65              ^diffutils@3.8%gcc@11.2.0 arch=cray-sles15-zen3
[+]  po4cjxv                  ^libiconv@1.16%gcc@11.2.0 libs=shared,static arch=cray-sles15-zen3
[+]  yqxusqu          ^gdbm@1.19%gcc@11.2.0 arch=cray-sles15-zen3
[+]  fdyfuat              ^readline@8.1.2%gcc@11.2.0 arch=cray-sles15-zen3
[+]  33sp27w                  ^ncurses@6.3%gcc@11.2.0~symlinks+termlib abi=none arch=cray-sles15-zen3
[+]  pysfq3q                      ^pkgconf@1.8.0%gcc@11.2.0 arch=cray-sles15-zen3
[+]  ymk6b64          ^zlib@1.2.12%gcc@11.2.0+optimize+pic+shared patches=0d38234 arch=cray-sles15-zen3
```

Compiler entry:
```
compilers:
  - compiler:
      spec: gcc@11.2.0
      paths:
        cc: gcc
        cxx: g++
        f77: gfortran
        fc: gfortran
      flags: {}
      operating_system: sles15
      target: any
      modules:
      - PrgEnv-gnu
      - gcc/11.2.0
      environment: {}
      extra_rpaths: []
```

Build error:
```
==> Installing openblas-0.3.21-3p4uqpakkiirevttvnortapbtczppcbl
==> No binary for openblas-0.3.21-3p4uqpakkiirevttvnortapbtczppcbl found: installing from source
==> Fetching https://github.com/xianyi/OpenBLAS/archive/v0.3.21.tar.gz
==> No patches needed for openblas
==> openblas: Executing phase: 'edit'
==> openblas: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'CC=/autofs/nccs-svm1_home1/eugeneswalker/openblas/spack/lib/spack/env/gcc/gcc' 'FC=/autofs/nccs-svm1_home1/eugeneswalker/openblas/spack/lib/spack/env/gcc/gfortran' 'MAKE_NB_JOBS=0' 'ARCH=x86_64' 'TARGET=ZEN' 'USE_LOCKING=1' 'USE_OPENMP=0' 'USE_THREAD=0' 'RANLIB=ranlib' 'libs' 'netlib' 'shared'

34 errors found in build log:
     2043    /autofs/nccs-svm1_home1/eugeneswalker/openblas/spack/lib/spack/env/gcc/gfortran -O2 -hnopattern -O noomp -fPIC -c -o sgebak.o sgebak.f
     2044    /autofs/nccs-svm1_home1/eugeneswalker/openblas/spack/lib/spack/env/gcc/gfortran -O2 -hnopattern -O noomp -fPIC -c -o sgebal.o sgebal.f
     2045    /autofs/nccs-svm1_home1/eugeneswalker/openblas/spack/lib/spack/env/gcc/gfortran -O2 -hnopattern -O noomp -fPIC -c -o sgebd2.o sgebd2.f
     2046    /autofs/nccs-svm1_home1/eugeneswalker/openblas/spack/lib/spack/env/gcc/gfortran -O2 -hnopattern -O noomp -fPIC -c -o sgebrd.o sgebrd.f
     2047    /autofs/nccs-svm1_home1/eugeneswalker/openblas/spack/lib/spack/env/gcc/gfortran -O2 -hnopattern -O noomp -fPIC -c -o sgecon.o sgecon.f
     2048    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2049    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2050    make[2]: *** [Makefile:66: sgbsv.o] Error 1
     2051    make[2]: *** Waiting for unfinished jobs....
     2052    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2053    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2054    make[2]: *** [Makefile:66: spotrf2.o] Error 1
     2055    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2056    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2057    make[2]: *** [Makefile:66: sgecon.o] Error 1
     2058    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2059    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2060    make[2]: *** [Makefile:66: sgetrf2.o] Error 1
     2061    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2062    gfortran: error: noomp: linker input file not found: No such file or directory
     2063    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2064    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2065    make[2]: *** [Makefile:66: sgbcon.o] Error 1
  >> 2066    make[2]: *** [Makefile:66: sgebak.o] Error 1
     2067    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2068    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2069    make[2]: *** [Makefile:66: sgbtrs.o] Error 1
     2070    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2071    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2072    make[2]: *** [Makefile:66: sgbtf2.o] Error 1
     2073    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2074    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2075    make[2]: *** [Makefile:66: sgbequ.o] Error 1
     2076    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2077    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2078    make[2]: *** [Makefile:66: sgebd2.o] Error 1
     2079    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2080    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2081    make[2]: *** [Makefile:66: sgebrd.o] Error 1
     2082    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2083    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2084    make[2]: *** [Makefile:66: sgebal.o] Error 1
     2085    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2086    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2087    make[2]: *** [Makefile:66: sgbrfs.o] Error 1
     2088    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2089    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2090    make[2]: *** [Makefile:66: sgbsvx.o] Error 1
     2091    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2092    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2093    make[2]: *** [Makefile:66: sgbbrd.o] Error 1
     2094    gfortran: warning: noomp: linker input file unused because linking not done
  >> 2095    gfortran: error: noomp: linker input file not found: No such file or directory
  >> 2096    make[2]: *** [Makefile:66: sgbtrf.o] Error 1
     2097    make[2]: Leaving directory '/tmp/eugeneswalker/spack-stage/spack-stage-openblas-0.3.21-3p4uqpakkiirevttvnortapbtczppcbl/spack-src/lapack-netlib/SRC'
  >> 2098    make[1]: *** [Makefile:27: lapacklib] Error 2
     2099    make[1]: Leaving directory '/tmp/eugeneswalker/spack-stage/spack-stage-openblas-0.3.21-3p4uqpakkiirevttvnortapbtczppcbl/spack-src/lapack-netlib'
  >> 2100    make: *** [Makefile:253: netlib] Error 2
```

### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>

### Information on your system

* **Spack:** 0.19.0.dev0 (cd015b84980bb26190c73c0cdbbcf0cc4b0db95e)
* **Python:** 3.9.7
* **Platform:** cray-sles15-zen3
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/9832972/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/9832973/spack-build-out.txt)

@martin-frbg @lukebroskop @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by martin-frbg** [comment]
Should be fixed by https://github.com/xianyi/OpenBLAS/pull/3778 - sorry for the inconvenience but I do not have access to any Cray system

**Comment by eugeneswalker** [comment]
> Should be fixed by [xianyi/OpenBLAS#3778](https://github.com/xianyi/OpenBLAS/pull/3778) - sorry for the inconvenience but I do not have access to any Cray system

Thank you! I will create a PR here that patches that in!
